main.tf

```diff
@@ -46,11 +38,7 @@ resource "azurerm_application_security_group" "vm" {
 }
 
 resource "azurerm_network_interface_application_security_group_association" "vm" {
-  # https://github.com/hashicorp/terraform/issues/10857
-  #
-  # NOTE: switch the count to `length(azurerm_network_interface.vm.*.id)` as
-  # soon as computed values are supported.
-  count = var.instance_count
```

Comment:
Should be a vestigial variable left over in variables.tf

Reply:
Still used to drive instance count